usersession/userd/settings.go

```diff
@@ -151,75 +265,83 @@ func (s *Settings) Get(setting string, sender dbus.Sender) (string, *dbus.Error)
 	}
 
 	cmd := exec.Command("xdg-settings", "get", setting)
-	output, err := cmd.CombinedOutput()
+	output, err := checkOutput(cmd, "get", setting)
 	if err != nil {
-		return "", dbus.MakeFailedError(fmt.Errorf("cannot get setting %s: %s", setting, osutil.OutputErr(output, err)))
+		return "", err
 	}
 
-	// avoid information leak: see https://github.com/snapcore/snapd/pull/4073#discussion_r146682758
-	snap, err := snapFromSender(s.conn, sender)
+	desktopFile, err := outputDesktopFile(s, output, sender)
 	if err != nil {
-		return "", dbus.MakeFailedError(err)
+		return "", err
 	}
-	if !strings.HasPrefix(string(output), snap+"_") {
-		return "NOT-THIS-SNAP.desktop", nil
+
+	return strings.TrimSpace(desktopFile), nil
+}
+
+// GetSub implements the 'GetSub' method of the 'io.snapcraft.Settings'
```

Comment:
Neither the name nor the comment really explain what this is about. It should really say something like
```
GetSub reads and returns a sub-property value by asking xdg-settings
```

Risking asking a silly question, why do we need `GetSub` if we have `Get`, can we somehow just extend the semantics so that get is sufficient?

Reply:
To answer my own question: please refer to `xdc-settings(1)` manual page. The system explicitly permits setting of a sub-property of a property. Therefore it this implementation matches that exactly.